### PR TITLE
Support multiple results in view + Tag recipes use case

### PR DESCRIPTION
**Pull request by jaeyonglee3**
This PR brings in several changes and new things:

- An overhauled result view (no longer using GridBagLayout) that supports up to 15 results at once.
- Connected the search by name + cuisine use cases to the result view (no longer printing results to console) by implementing their view models, presenters, and editing their respective views.
- Fully implemented the ability for users to tag recipes.
- Edited the Main.java and ViewManagerModel.java files so that the main JFrame adapts to the dimensions specified by the panels within them (using setPreferredSize), and not vice versa.

### CODE REVIEW

File: src/use_case/tag_recipe/TagRecipeInputData.java
Comment (pratzyushd) [inline src/use_case/tag_recipe/TagRecipeInputData.java:1]: The getter methods in this class should have access modifiers.

File: src/use_case/tag_recipe/TagRecipeInputBoundary.java
Comment (pratzyushd) [inline src/use_case/tag_recipe/TagRecipeInputBoundary.java:1]: As before, add `public`.

**Review by pratzyushd — CHANGES_REQUESTED** [review]
Generally well written code, making good use of design patterns.

There are some fixes to be made, largely to do with access modifiers being included in places. Note that while not required in interfaces, we probably should include them in the interfaces just for the sake of consistency and reducing mental overhead when looking through the code.

**Comment by pratzyushd** [comment]
I believe that this PR should close #11 as well?

File: src/interface_adapter/search_by_name/SearchByNamePresenter.java [hunk 1]
```diff
@@ -33,6 +41,34 @@ public void prepareSuccessView(SearchNameOutputData response) {
         for (Map<String, String> recipe : response.getRecipes()) {
             System.out.println(recipe.get("name"));
         }
+
```
[the remark this answers is not in the document]
Comment (jaeyonglee3) [reply src/interface_adapter/search_by_name/SearchByNamePresenter.java:44]: Yes, I've noticed that too. This should definitely be something for us to consider down the line.

**Review by pratzyushd — APPROVED** [review]
LGTM, great job! :+1:

**Pull request merged**